app/controllers/api/v1/targets_controller.rb

```diff
@@ -7,12 +7,21 @@ def index
 
       def create
         authorize Target
-        @target = TargetService.new(current_user, resource_params).create!
+        @target = TargetService.new(current_user, create_resource_params).create!
       end
 
-      def resource_params
+      def destroy
+        authorize Target
+        @target = TargetService.new(current_user, delete_resource_params).destroy!
```

Comment:
I think you are over-engineering this, you can still have the `resource_params` as you had for the create, but here just do:
```suggestion
        @target = TargetService.new(current_user, params[:id]).destroy!
```
you may ask, why don't using require and permit, that is a safety measure? well, here you only need the `id` and using `params[:id]` you are explicitly saying which param value you are accepting, and ignoring any other parameter

Reply:
I like the idea. In the first case, I did it this way because I thought about this question: "why don't using require and permit, that is a safety measure?" but your response make sense for me.